### ISSUE TITLE
[a11y]:  [Windows High Contrast Mode]: CopyButton, Close Button in InlineNotification with 'lowContrast=true' and IconButton with kind='ghost' are not visible in windows 10 High-Contrast Mode

### ISSUE THREAD

**Issue by SushanthTanariIBM** [carbon-design-system/carbon#19015]
### Package

@carbon/react

### Browser

Chrome

### Operating System

Windows

### Package version

1.79.0

### React version

18.0.0

### Automated testing tool and ruleset

Manual Verification

### Assistive technology

NVDA

### Description

"CopyButton", "Close Button in InlineNotification with lowContrast=true" and "IconButton with kind='ghost'" are not visible in windows 10 High-Contrast Mode

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-iconbutton--default&args=kind:ghost

### Steps to reproduce

Urls to @carbon/react storybook -

https://react.carbondesignsystem.com/?path=/docs/components-copybutton--overview&args=kind:ghost
https://react.carbondesignsystem.com/?path=/story/components-iconbutton--default&args=kind:ghost
https://react.carbondesignsystem.com/?path=/story/components-notifications-inline--default&args=lowContrast:!true

IconButton:

![Image](https://github.com/user-attachments/assets/4d088a81-d7ca-48ed-9701-d54f2cfede1a)

CopyButton:

![Image](https://github.com/user-attachments/assets/4cc7f1ce-dcec-47e7-b917-794691e4dc59)

"Close Button in InlineNotification with lowContrast=true" :
![Image](https://github.com/user-attachments/assets/5a59f53c-34b0-4752-a80b-fec358c8644f)

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by wkeese** [comment]
I'm looking into this.

I reproduce this (or at least some of it) on Windows 11 too, using the "Aquatic" theme.  Note that you can work around it by setting your Carbon theme to G100.  

Basically, it's a CSS precedence problem.   When you are in a high-contrast theme, you need to set your `fill` to a system color.  And Carbon does that twice for the Copy icon via:

```css
@media screen and (-ms-high-contrast: active) {
    svg {
        fill: ButtonText;
    }
}

...

@media screen and (-ms-high-contrast: active), (forced-colors: active) {
    .cds--snippet__icon {
        fill: ButtonText;
    }
}
```

However, both CSS rules above get overridden by Copy button's standard (non-high-contrast-related) CSS:

```css
.cds--copy-btn svg {
    fill: var(--cds-icon-primary, #161616);
}
```

Likewise for the IconButton, Carbon has the correct CSS:

```css
@media screen and (-ms-high-contrast: active) {
    svg {
        fill: ButtonText;
    }
}
```

But it's overridden by the very specific (and non-high-contrast related):

```css
.cds--btn--ghost:not([disabled]) svg {
    fill: var(--cds-icon-primary, #161616);
}
```

**Comment by heloiselui** [comment]
This was fixed in PR https://github.com/carbon-design-system/carbon/pull/19352
